en/debug/probe_stlink.md

```diff
@@ -0,0 +1,197 @@
+
+The GPIO1/2 pins are not supported by the STLinkv3, and we recommend using digital ITM profiling over SWO which is much more flexible and supports cycle accurate timestamping.
+
+You can choose to solder a short or long cable to the BTB connector.
+The short cable is better for high-speed communication, but is more difficult to solder.
+We recommend soldering the long cable first and testing how fast you can communicate with your target.
+
+:::note
+This guide is written for the full 10-pin debug port.
+If you want to solder the mini 6-pin version, just leave out the signals you don't have.
+The STLink supports any SWD/JTAG-based debug interface, so you can adapt this guide for any other connector you may have.
+The debug probes are so cheap, you can just have one per connector instead of using adapters.
+:::
+
+This is how the STLinkv3-MINIE is delivered.
+
+![](../../assets/debug/stlinkv3_minie_p1.jpeg)
+
+Unwrap the PCB and check it for any damage.
+Plug it in and see if it powers on correctly.
+
+![](../../assets/debug/stlinkv3_minie_p2.jpeg)
+
+### Short Cable
+
+The short cable requires a wire cutter and stripper and requires a little more soldering skill.
+However, it makes the entire debug probe even smaller.
+
+Assemble a 10-pin connector without GPIO1/2. If you already have an assembled cable, carefully remove the two GPIO1/2 cables with a tweezer by lifting the pegs that keep the cables secured.
+Cut the cables to a short ~2cm (~1in) length and strip the wires.
+
+![](../../assets/debug/stlinkv3_minie_p3.jpeg)
+
+Tin both the BTB connector on the STLink and the cables.
+
+![](../../assets/debug/stlinkv3_minie_p4.jpeg)
+
+First solder the GND and VCC signals to align the connector in parallel to the edge.
+Then solder the TX and RX pins. Solder the RST connection last.
+
+![](../../assets/debug/stlinkv3_minie_p5.jpeg)
+
+Turn the STLink over and solder the remaining three wires.
+Start with SWDIO->TMS, then SWCLK->CLK, and finally SWO->TDO.
+
+![](../../assets/debug/stlinkv3_minie_p6.jpeg)
+
+### Long Cable
+
+The long cable is particularly useful if you use pre-assembled cables as it removes the need to cut wires or strip them.
+
+Carefully remove the two GPIO1/2 cables from one connector of the cable.
+Then remove all cables from the other connector.
+You are left with a eight crimped connectors at the end of the wires.
+
+![](../../assets/debug/stlinkv3_minie_p7.jpeg)
+
+Tin the crimped cable connectors and BTB connector and solder the crimped connectors directly to the STLinkv3.
+Be careful to not create shorts between the cables, as the crimped connectors are quite large.
+
+![](../../assets/debug/stlinkv3_minie_p8.jpeg)
+
+### Testing
+
+You should now test your debug probe to ensure you do not have any electrical shorts.
+
+1. Plug the probe into your target via the Pixhawk Debug Port.
+2. Test the serial port with a program of your choice.
+3. Test the SWD and RST connection via [OpenOCD][] or [STLink][] software.
```

Comment:
Missing references

Reply:
Empty references - linked up using your references at the end, which only had one item.

```suggestion
3. Test the SWD and RST connection via [OpenOCD][https://openocd.org] or [STLink](https://www.st.com/en/development-tools/stsw-link004.html) software.
```